extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts

```diff
@@ -367,48 +380,99 @@ export class DataExtensionsEditorView extends AbstractWebview<
       message: "Retrieving usages",
     });
 
-    const usages = await getAutoModelUsages({
-      cliServer: this.cliServer,
-      queryRunner: this.queryRunner,
-      queryStorageDir: this.queryStorageDir,
-      databaseItem: this.databaseItem,
-      progress: (update) => this.showProgress(update, maxStep),
-    });
+    let predictedModeledMethods: Record<string, ModeledMethod>;
 
-    await this.showProgress({
-      step: 1800,
-      maxStep,
-      message: "Creating request",
-    });
+    if (useLlmGenerationV2()) {
+      const usages = await runAutoModelQueries(this.mode, {
+        cliServer: this.cliServer,
+        queryRunner: this.queryRunner,
+        queryStorageDir: this.queryStorageDir,
+        databaseItem: this.databaseItem,
+        progress: (update) => this.showProgress(update, maxStep),
+      });
+      if (!usages) {
+        return;
+      }
 
-    const request = createAutoModelRequest(
-      this.databaseItem.language,
-      externalApiUsages,
-      modeledMethods,
-      usages,
-      this.mode,
-    );
+      await this.showProgress({
+        step: 1800,
+        maxStep,
+        message: "Creating request",
+      });
 
-    await this.showProgress({
-      step: 2000,
-      maxStep,
-      message: "Sending request",
-    });
+      const request = await createAutoModelV2Request(this.mode, usages);
 
-    const response = await this.callAutoModelApi(request);
-    if (!response) {
-      return;
-    }
+      await this.showProgress({
+        step: 2000,
+        maxStep,
+        message: "Sending request",
+      });
 
-    await this.showProgress({
-      step: 2500,
-      maxStep,
-      message: "Parsing response",
-    });
+      const response = await this.callAutoModelApiV2(request);
+      if (!response) {
+        return;
+      }
 
-    const predictedModeledMethods = parsePredictedClassifications(
-      response.predicted || [],
-    );
+      await this.showProgress({
+        step: 2500,
+        maxStep,
+        message: "Parsing response",
+      });
+
+      const models = loadYaml(response.models, {
+        filename: "auto-model.yml",
+      });
+
+      const modeledMethods = loadDataExtensionYaml(models);
```

Comment:
I just want to double check: this replaces all modeled methods right? But that's okay because we only have a button that models the whole package and we have 1 YAML per package.

If we want to add a button for a single candidate then we'd need to merge the existing modeled methods and any that we get from automodel.

Reply:
It does not replace all modeled methods, it will already merge the modeled methods. Anything returned from the LLM will any already existing modeled methods, but if something is not returned from the LLM, anything the user has already modeled will be unchanged.